mm2src/coins_activation/src/z_coin_activation.rs

```diff
@@ -0,0 +1,227 @@
+use crate::context::CoinsActivationContext;
+use crate::prelude::*;
+use crate::standalone_coin::{InitStandaloneCoinActivationOps, InitStandaloneCoinError,
+                             InitStandaloneCoinInitialStatus, InitStandaloneCoinTaskHandle,
+                             InitStandaloneCoinTaskManagerShared};
+use async_trait::async_trait;
+use coins::coin_balance::{EnableCoinBalance, IguanaWalletBalance};
+use coins::utxo::rpc_clients::ElectrumRpcRequest;
+use coins::utxo::{UtxoActivationParams, UtxoRpcMode};
+use coins::z_coin::{z_coin_from_conf_and_params, ZCoin, ZCoinBuildError};
+use coins::{BalanceError, CoinProtocol, MarketCoinOps, PrivKeyBuildPolicy, RegisterCoinError};
+use common::executor::Timer;
+use common::mm_ctx::MmArc;
+use common::mm_error::prelude::*;
+use crypto::hw_rpc_task::{HwRpcTaskAwaitingStatus, HwRpcTaskUserAction};
+use derive_more::Display;
+use futures::compat::Future01CompatExt;
+use rpc_task::RpcTaskError;
+use ser_error_derive::SerializeErrorType;
+use serde_derive::{Deserialize, Serialize};
+use serde_json::Value as Json;
+use std::time::Duration;
+
+pub type ZcoinTaskManagerShared = InitStandaloneCoinTaskManagerShared<ZCoin>;
+pub type ZcoinRpcTaskHandle = InitStandaloneCoinTaskHandle<ZCoin>;
+pub type ZcoinAwaitingStatus = HwRpcTaskAwaitingStatus;
+pub type ZcoinUserAction = HwRpcTaskUserAction;
+
+#[derive(Clone, Serialize)]
+pub struct ZcoinActivationResult {
+    pub current_block: u64,
+    pub wallet_balance: EnableCoinBalance,
+}
+
+impl GetCurrentBlock for ZcoinActivationResult {
+    fn get_current_block(&self) -> u64 { self.current_block }
+}
+
+#[derive(Clone, Serialize)]
+pub enum ZcoinInProgressStatus {
```

Comment:
I think we should use #[non_exhaustive] attribute for the public enums in whole project. It increases the compatibility of project in general. Enforces people from outside of the crate to handle _ in their match statements.

I explain that also for `Lightning channels and payments history` PR.